### ISSUE TITLE
Tags: Have a clearer indicator that a tag modification is changed

### ISSUE THREAD

**Issue by jeremeylduvall** [wordpress-mobile/WordPress-iOS#8471]
When I’m editing a tag, I’m a bit wary about clicking the “< Tags” navigation because I feel like my work isn’t saved although it is.

### Expected behavior
I expected to have two options at the top of the navigation:

- Close - Return to previous screen
- Save - Save my work (inactive if current work is already saved)

Then, the trash/delete could be underneath the tag description.

### Actual behavior
I have either a trash can icon or the option for "< Tags" which returns me to the main screen. It's not clear that your work is being automatically saved so you can click "< Tags" without losing anything.

![img_4370](https://user-images.githubusercontent.com/7240478/35161274-e9470aa0-fcfc-11e7-81c0-b0ae6d10a58d.png)

**Comment by elibud** [comment]
@ctarda @mattmiklic I think this works like this by design right? Most screen on the iOS app (not the Editor) save the changes without a prompt so I'm not sure we'll want to change that behavior here. This is working consistently with the rest of site settings.

**Comment by mattmiklic** [comment]
Yeah, this follows the pattern used in the iOS Settings app as well, where tapping Back automatically saves any changes made to a text input setting.

One thing I just noticed is that tapping "Done" on the keyboard doesn't have any effect on the tag editing screen. Maybe it should advance the focus to the Description field, and then tapping it again should close the screen? I noticed that tapping Done on a system-level settings screen (try renaming your iPhone from General > About) does this.
